com.hazelcast.jet.sql.impl.SqlResultImplTest.when_hasNextInterrupted_then_interrupted

_master_ (commit 7e7f00be7b5ed17b2afce73b5b14598e45350175)

Failed on ZingJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZingJDK8/384/testReport/com.hazelcast.jet.sql.impl/SqlResultImplTest/when_hasNextInterrupted_then_interrupted/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at com.hazelcast.jet.sql.impl.SqlResultImplTest.lambda$when_hasNextInterrupted_then_interrupted$2(SqlResultImplTest.java:74)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1221)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1323)
	at com.hazelcast.jet.sql.impl.SqlResultImplTest.when_hasNextInterrupted_then_interrupted(SqlResultImplTest.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output (hiccups are really high):</summary>

```
11:34:54,904 ERROR || - [EventServiceImpl] hz.hungry_wilbur.event-260 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] hz.hungry_wilbur.event-260 caught an exception while processing:com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher@77d7510
java.lang.UnsupportedOperationException: SQL catalog must never be cleared - MapEvent{entryEventType=CLEAR_ALL, member=Member [127.0.0.1]:5701 - 9e2ef0fe-f5ae-4b7e-8e76-b85515574b88 this, name='__sql.catalog', numberOfEntriesAffected=0}
	at com.hazelcast.jet.sql.impl.schema.TablesStorage$EntryListenerAdapter.mapCleared(TablesStorage.java:202) ~[classes/:?]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapEventPublishingService.dispatchEvent(ReplicatedMapEventPublishingService.java:132) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.replicatedmap.impl.ReplicatedMapService.dispatchEvent(ReplicatedMapService.java:232) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.eventservice.impl.LocalEventDispatcher.run(LocalEventDispatcher.java:64) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:245) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:228) [hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
11:34:54,905 ERROR |when_hasNextInterrupted_then_interrupted| - [SqlResultImplTest] Thread-716 - Unexpected exception caught
com.hazelcast.sql.HazelcastSqlException: The Jet SQL job failed: 
	at com.hazelcast.sql.impl.QueryUtils.toPublicException(QueryUtils.java:72) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.sql.impl.SqlResultImpl$RowToSqlRowIterator.hasNext(SqlResultImpl.java:131) ~[classes/:?]
	at com.hazelcast.jet.sql.impl.SqlResultImplTest.lambda$when_hasNextInterrupted_then_interrupted$1(SqlResultImplTest.java:62) ~[test-classes/:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0-zing_18.06.0.0]
Caused by: com.hazelcast.sql.impl.QueryException: The Jet SQL job failed: 
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:65) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.sql.impl.PlanExecutor.lambda$execute$3(PlanExecutor.java:355) ~[classes/:?]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977) ~[?:1.8.0-zing_18.06.0.0]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.AbstractJobProxy$CallbackBase.accept(AbstractJobProxy.java:367) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.AbstractJobProxy$CallbackBase.accept(AbstractJobProxy.java:350) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.spi.impl.AbstractInvocationFuture$WhenCompleteNode.lambda$execute$0(AbstractInvocationFuture.java:1563) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.test.TestLoggingUtils$CustomTestNameAwareForkJoinPool$TestNameAwareRunnable.run(TestLoggingUtils.java:104) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[?:1.8.0-zing_18.06.0.0]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[?:1.8.0-zing_18.06.0.0]
Caused by: java.util.concurrent.CancellationException
	at java.util.concurrent.CompletableFuture.cancel(CompletableFuture.java:2263) ~[?:1.8.0-zing_18.06.0.0]
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:294) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:692) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.cancelAllExecutions(JobExecutionService.java:238) ~[hazelcast-5.1-SNAPSHOT.jar:5.1-SNAPSHOT]
	at com.hazelcast.jet.SimpleTestInClusterSupport.supportAfter(SimpleTestInClusterSupport.java:102) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at sun.reflect.GeneratedMethodAccessor100.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0-zing_18.06.0.0]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0-zing_18.06.0.0]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:381) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:62) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:49) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:162) ~[hazelcast-5.1-SNAPSHOT-tests.jar:5.1-SNAPSHOT]
	... 1 more
11:34:54,951 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:34:59,951 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:04,952 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:09,952 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:14,874  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.1G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=380.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=88.59%, heap.memory.used/max=88.59%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=35.71%, load.system=58.33%, load.systemAverage=3.74, thread.count=47, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:35:14,952 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-9 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:19,953 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:24,953 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:29,954 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:34,876  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.1G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=372.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=88.83%, heap.memory.used/max=88.83%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=30.77%, load.system=53.85%, load.systemAverage=3.24, thread.count=48, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:35:34,954 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:39,955 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:44,955 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:49,956 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:54,877  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.0G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=368.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=88.95%, heap.memory.used/max=88.95%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=28.57%, load.system=53.85%, load.systemAverage=3.03, thread.count=49, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:35:54,956 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:35:59,957 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-12 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:04,957 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:09,957 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:14,879  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.0G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=362.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=89.13%, heap.memory.used/max=89.13%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=26.67%, load.system=53.33%, load.systemAverage=2.73, thread.count=48, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:36:14,958 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:19,959 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:24,959 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:29,960 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-4 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:34,880  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.0G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=362.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=89.13%, heap.memory.used/max=89.13%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=26.67%, load.system=53.33%, load.systemAverage=2.53, thread.count=48, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:36:34,960 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:39,960 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:44,961 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-12 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:49,961 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:54,882  INFO || - [HealthMonitor] hz.hungry_wilbur.HealthMonitor - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] processors=4, physical.memory.total=29.8G, physical.memory.free=10.0G, swap.space.total=0, swap.space.free=0, heap.memory.used=2.9G, heap.memory.free=358.0M, heap.memory.total=3.3G, heap.memory.max=3.3G, heap.memory.used/total=89.25%, heap.memory.used/max=89.25%, minor.gc.count=0, minor.gc.time=0ms, major.gc.count=0, major.gc.time=0ms, unknown.gc.count=44, unknown.gc.time=22712ms, load.process=40.00%, load.system=50.00%, load.systemAverage=2.38, thread.count=48, thread.peakCount=152, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=9, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
11:36:54,962 DEBUG || - [JobRepository] hz.hungry_wilbur.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Job cleanup took 0ms
11:36:55,036  INFO |when_hasNextInterrupted_then_interrupted| - [SqlTestSupport] Thread-715 - Removing 0 cached plans in SqlTestSupport.@After
11:36:55,038  INFO |when_hasNextInterrupted_then_interrupted| - [SimpleTestInClusterSupport] Thread-715 - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
11:36:55,038  INFO |when_hasNextInterrupted_then_interrupted| - [SimpleTestInClusterSupport] Thread-715 - Destroying 0 distributed objects in SimpleTestInClusterSupport.@After: []
BuildInfo right after when_hasNextInterrupted_then_interrupted(com.hazelcast.jet.sql.impl.SqlResultImplTest): BuildInfo{version='5.1-SNAPSHOT', build='20220119', buildNumber=20220119, revision=7e7f00b, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'when_hasNextInterrupted_then_interrupted(com.hazelcast.jet.sql.impl.SqlResultImplTest):'
11:34:50, accumulated pauses: 4948 ms, max pause: 20 ms, pauses over 1000 ms: 0
11:34:55, accumulated pauses: 4949 ms, max pause: 13 ms, pauses over 1000 ms: 0
11:35:00, accumulated pauses: 4939 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:35:05, accumulated pauses: 4949 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:35:10, accumulated pauses: 4947 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:35:15, accumulated pauses: 4942 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:35:20, accumulated pauses: 4948 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:35:25, accumulated pauses: 4943 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:35:30, accumulated pauses: 4944 ms, max pause: 15 ms, pauses over 1000 ms: 0
11:35:35, accumulated pauses: 4949 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:35:40, accumulated pauses: 4943 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:35:45, accumulated pauses: 4953 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:35:50, accumulated pauses: 4939 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:35:55, accumulated pauses: 4944 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:36:00, accumulated pauses: 4946 ms, max pause: 13 ms, pauses over 1000 ms: 0
11:36:05, accumulated pauses: 4951 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:36:10, accumulated pauses: 4950 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:36:15, accumulated pauses: 4941 ms, max pause: 12 ms, pauses over 1000 ms: 0
11:36:20, accumulated pauses: 4950 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:36:25, accumulated pauses: 4945 ms, max pause: 13 ms, pauses over 1000 ms: 0
11:36:30, accumulated pauses: 4938 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:36:35, accumulated pauses: 4947 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:36:40, accumulated pauses: 4947 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:36:45, accumulated pauses: 4948 ms, max pause: 11 ms, pauses over 1000 ms: 0
11:36:50, accumulated pauses: 4949 ms, max pause: 10 ms, pauses over 1000 ms: 0
11:36:55, accumulated pauses: 20 ms, max pause: 10 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlResultImplTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlResultImplTest.java
@@ -20,7 +20,7 @@ import com.hazelcast.function.ConsumerEx;
 import com.hazelcast.jet.sql.SqlTestSupport;
 import com.hazelcast.sql.SqlResult;
 import com.hazelcast.sql.SqlRow;
-import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.junit.BeforeClass;
@@ -34,7 +34,7 @@ import java.util.concurrent.atomic.AtomicBoolean;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.Assert.assertTrue;
 
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class SqlResultImplTest extends SqlTestSupport {
 

